Fixing text color issue after an error

rocknes opened the pull request:
Also exposing option to turn off colored output entirely.

Tested manually to delete a non existent independent disk,
Before fix : After error console text color became dark gray
After fix : After error console text color remained the same as the foreground color

Also checked that setting the environment variable USE_COLROED_OUTPUT to anything other than '1' lead to error messages being printed out in normal text color. If the environment variable is unset or set to '1' color RED is used to print error messages.

Tested the following scenarios

Environment variable USE_COLORED_OUPUT not set
------------------------------------------------------
vcd info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Colored output
vcd --colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Colored output
vcd --no-colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Monochrome output

Environment variable USE_COLORED_OUPUT is set to 0
------------------------------------------------------
vcd info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 :  Monochrome output
vcd --colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Colored output
vcd --no-colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Monochrome output

Environment variable USE_COLORED_OUPUT is set to 1
------------------------------------------------------
vcd info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 :  Colored output
vcd --colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Colored output
vcd --no-colorized info catalog 87d6f1f7-6d29-4abc-9f09-ce8322175901 : Monochrome output


<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/vcd-cli/145)
<!-- Reviewable:end -->

pacogomez commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vcd_cli/utils.py, line 206 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0ir1NZ1wPRtuH1iH9T:-L0ir1NZ1wPRtuH1iH9U:b-fvobvo) ([raw file](https://github.com/vmware/vcd-cli/blob/e351286d63904c30427aaf8c7b103502ec542a27/vcd_cli/utils.py#L206)):*
> ```Python
>         sys.exit(1)
>     else:
>         if 'USE_COLORED_OUTPUT' not in environ.keys() or \
> ```

we need to enable/disable colored output of `info` command, so when redirecting output to a file, the ansii escape characters are not included. The couple of calls to `highlight` https://github.com/vmware/vcd-cli/blob/e8cbd505734e27dc85fc83502896cf56c01f8c10/vcd_cli/utils.py#L131 and https://github.com/vmware/pyvcloud/blob/067ba70dd814eca4520053a8dac72410e31c068b/pyvcloud/vcd/utils.py#L377

I think we should add a top level option at the `vcd` level, like `--colorized/--no-colorized` with a corresponding `VCD_COLORIZED` env variable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

anusuyar commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[vcd_cli/utils.py, line 206 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0ir1NZ1wPRtuH1iH9T:-L0lD83f2X4aJQvvDhk6:b-fcqwu8) ([raw file](https://github.com/vmware/vcd-cli/blob/e351286d63904c30427aaf8c7b103502ec542a27/vcd_cli/utils.py#L206)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

we need to enable/disable colored output of `info` command, so when redirecting output to a file, the ansii escape characters are not included. The couple of calls to `highlight` https://github.com/vmware/vcd-cli/blob/e8cbd505734e27dc85fc83502896cf56c01f8c10/vcd_cli/utils.py#L131 and https://github.com/vmware/pyvcloud/blob/067ba70dd814eca4520053a8dac72410e31c068b/pyvcloud/vcd/utils.py#L377

I think we should add a top level option at the `vcd` level, like `--colorized/--no-colorized` with a corresponding `VCD_COLORIZED` env variable
</blockquote></details>

Made the changes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: 0 of 3 files reviewed at latest revision, 6 unresolved discussions.

---

*[vcd_cli/utils.py, line 132 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lX0iTn6PCLE4ez0Zm:-L0lX0iTn6PCLE4ez0Zn:bu53klc) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L132)):*
> ```Python
>         if sys.version_info[0] < 3:
>             text = str(text, 'utf-8')
>         if ctx.find_root().params['is_colorized'] is True:
> ```

`is True` is redundant

---

*[vcd_cli/utils.py, line 204 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lX7xKKjMF45SBNjyk:-L0lX7xKKjMF45SBNjyl:bu53klc) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L204)):*
> ```Python
>         if sys.version_info[0] < 3:
>             text = str(text, 'utf-8')
>         if ctx.find_root().params['is_colorized'] is True:
> ```

`is True` is redundant

---

*[vcd_cli/utils.py, line 214 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXFAk6u61dRTAch_X:-L0lXFAk6u61dRTAch_Y:byr4ffv) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L214)):*
> ```Python
>         click.echo('\x1b[2K\r', nl=False)
>         if ctx is not None:
>             if ctx.find_root().params['is_colorized'] is True:
> ```

see above

---

*[vcd_cli/utils.py, line 218 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXH0xFTFqAztfqx1J:-L0lXH0xFTFqAztfqx1K:byrq2m7) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L218)):*
> ```Python
>             ctx.fail(message)
>         else:
>             if 'USE_COLORED_OUTPUT' in environ.keys() and \
> ```

see below

---

*[vcd_cli/vcd.py, line 18 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXKP41e-_loE7a0nL:-L0lXKP41e-_loE7a0nM:b-e68a4j) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L18)):*
> ```Python
> import pkg_resources
> import platform
> from colorama import init
> ```

imports should go in alphabetical order, run `flake8` before commit

---

*[vcd_cli/vcd.py, line 53 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lWc5PvfnE_XC4VwKo:-L0lWc5PvfnE_XC4VwKp:b1ju2kx) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L53)):*
> ```Python
>               'is_colorized',
>               default=True,
>               envvar='USE_COLORED_OUTPUT',
> ```

please prepend env var with `VCD_` to reduce possibility of name conflict.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
see comments

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-:-L0lXedrG4R6TC8l8hg8:bqud5t2)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[vcd_cli/vcd.py, line 53 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lWc5PvfnE_XC4VwKo:-L0laGdeLUzsVmFkwXVj:b-sdg7t0) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L53)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

please prepend env var with `VCD_` to reduce possibility of name conflict.
</blockquote></details>

also, in the `vcd` top level command, add a `Environment Variables` section in the help docstring. See `vcd login` command for an example

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[vcd_cli/utils.py, line 132 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lX0iTn6PCLE4ez0Zm:-L0lZrpH96FVWNW45Szt:ba3dfob) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L132)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

`is True` is redundant
</blockquote></details>

Removed

---

*[vcd_cli/utils.py, line 204 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lX7xKKjMF45SBNjyk:-L0lZtcJdZPhTOa7yNxx:ba3dfob) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L204)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

`is True` is redundant
</blockquote></details>

Removed

---

*[vcd_cli/utils.py, line 214 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXFAk6u61dRTAch_X:-L0l_-Z_19vovSWl5ywY:ba3dfob) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L214)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

see above
</blockquote></details>

Removed

---

*[vcd_cli/utils.py, line 218 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXH0xFTFqAztfqx1J:-L0l_3vu6gGFpUjEIyiq:b-uo8k0s) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/utils.py#L218)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

see below
</blockquote></details>

Renamed to VCD_USE_COLORED_OUTPUT

---

*[vcd_cli/vcd.py, line 18 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXKP41e-_loE7a0nL:-L0l_9LpVais0gE6vgkr:b5x1uik) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L18)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

imports should go in alphabetical order, run `flake8` before commit
</blockquote></details>

I did run flake8 and it didn't give any warnings. 
I have installed flake8_import_order module to make sure that future mistakes are avoided :)

---

*[vcd_cli/vcd.py, line 53 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lWc5PvfnE_XC4VwKo:-L0lhYIgv57fDlP8EJlr:bb6un1j) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L53)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

also, in the `vcd` top level command, add a `Environment Variables` section in the help docstring. See `vcd login` command for an example
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vcd_cli/vcd.py, line 18 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXKP41e-_loE7a0nL:-L0lnWtwj_3P7mlHgL95:b8vdjoq) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L18)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

I did run flake8 and it didn't give any warnings. 
I have installed flake8_import_order module to make sure that future mistakes are avoided :)

</blockquote></details>

if that is a new dependency, please add to `test-requirements.txt`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[vcd_cli/vcd.py, line 18 at r2](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-L0lXKP41e-_loE7a0nL:-L0luyxFj_evWEhzBRPJ:b-8b11j9) ([raw file](https://github.com/vmware/vcd-cli/blob/3a017d2f0168964dc88b564bad2b8af5144e06ff/vcd_cli/vcd.py#L18)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

if that is a new dependency, please add to `test-requirements.txt`
</blockquote></details>

Added

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-:-L0lxABzFEFVid1vM_pO:bnfp4nl)*
<!-- Sent from Reviewable.io -->

anusuyar commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/145#-:-L0pZl6ow_IURdGNDdX4:bnfp4nl)*
<!-- Sent from Reviewable.io -->
